### ISSUE TITLE
"could not resolve macro" in a working crate

### ISSUE THREAD

**Issue by osa1** [rust-analyzer/rust-analyzer#6950]
rust-analyzer version: df3652b66

In a crate that builds and runs fine (using latest stable release) rust-analyzer raises an error. crate code:

<details>

<summary>Cargo.toml</summary>

```toml
[package]
name = "macro_test"
version = "0.1.0"
authors = []
edition = "2018"

[dependencies]
maplit = "1.0.2"
```

</details>

<details>

<summary>main.rs</summary>

```rust
use maplit::hashmap;
use std::collections::HashMap;

fn main() {
    let _: HashMap<usize, Vec<usize>> = hashmap! {
        1 => vec![]
    };
}
```

</details>

rust-analyzer error:

![Screenshot_2020-12-19_17-59-05](https://user-images.githubusercontent.com/448274/102692412-172ccd00-4224-11eb-8117-cf4b373accf6.png)

**Comment by osa1** [comment]
I think this could possibly be a regression. I just realized that I started get a lot of macro-related errors after updating rust-analyzer (I update from master branch weekly). Here's another example that previously did not raise an error:

![Screenshot_2020-12-20_08-27-40](https://user-images.githubusercontent.com/448274/102705876-50555380-429d-11eb-8772-7a425e0b2ede.png)

The shown line does not even use a macro, I think the error is coming from the previous line (which may be another bug).

**Comment by osa1** [comment]
b1b7727e0 is a good commit where none of the code above raises an error.

**Comment by osa1** [comment]
git bisect output:

```
be50908a5031fee45fcbba97bc0c96187324ea54 is the first bad commit
commit be50908a5031fee45fcbba97bc0c96187324ea54
Author: Jonas Schievink <jonasschievink@gmail.com>
Date:   Mon Nov 30 20:26:35 2020 +0100

    Emit macro diagnostics when lowering bodies

 crates/hir_def/src/body.rs             | 73 +++++++++++++++++++++++-----------
 crates/hir_def/src/body/diagnostics.rs |  6 ++-
 crates/hir_def/src/body/lower.rs       | 14 ++++++-
 crates/hir_def/src/data.rs             |  2 +-
 4 files changed, 68 insertions(+), 27 deletions(-)
```

be50908a5031fee45fcbba97bc0c96187324ea54

**Comment by lnicola** [comment]
The macros never worked, only the diagnostics are new. You can disable the diagnostics, but issue reports are also valuable (take a look at the A-macro issues, both open and closed).

The futures one is a duplicate, the maplit error is new.

**Comment by osa1** [comment]
In case anyone else having the same problem, setting `rust-analyzer.diagnostics.enableExperimental` to `false` disables this buggy diagnostic.

**Comment by edwin0cheng** [comment]
Related  #6788

**Comment by lnicola** [comment]
Probably duplicate of #6788.

**Comment by la10736** [comment]
Or 
```
"rust-analyzer.diagnostics.disabled": [
        "macro-error"
    ]
```
